tmt/steps/finish/__init__.py

```diff
@@ -135,6 +149,24 @@ def go(self) -> None:
             guest.stop()
             guest.remove()
 
+        # Prune all irrelevant files and dirs
+        if not (self.opt('keep') or self.opt('dry')):
+            self.verbose('prune', f'Pruning TMT tree {self.plan.worktree}', color='red', shift=2)
+            shutil.rmtree(self.plan.worktree)
+            self.verbose(
+                'prune',
+                f'Pruning all non preserved files from {self.plan.workdir} '
+                f'except files containing one of {self._preserved_files}.',
+                color='red',
+                shift=2)
+            # If item in plan workdir does not contain a string from
+            # preserved files list, delete it recursively.
+            self.run(
+                f'find {self.plan.workdir} -mindepth 2 -maxdepth 2 ' +
+                ' '.join([f'| grep -v {x}' for x in self._preserved_files]) +
+                '| xargs rm -rvf',
+                shell=True)
```

Comment:
I'm not sure if grepping for `yaml` and `data` is the best approach here. The directories and fetched git repositories can contain a bunch of such files (including the `tmt` repo itself).

I was thinking about a `clean()` method which each `Step` would implement and which would take care of the cleanup and keeping the interesting files around. Or, maybe, `BasePlugin.cleanup()` method which would by default do a `shutil.rmtree(self.workdir)`?

@adiosnb, what do you think?

Reply:
I will add a generic clean() method and extends it if necessary to preserve custom files.

Reply:
Thanks, this looks much better now!